src/color.rs

```diff
@@ -329,65 +389,142 @@ impl<A: Copy> FromColor<A> for A {
     }
 }
 
-/// `FromColor` for Luma
+/// Copy-based conversions to target pixel types using `FromColor`.
+pub(crate) trait IntoColor<Other> {
```

Comment:
Can you add a huge `FIXME: remove this trait`? Reinforcing the belief of `RGB == sRGB` is potentially harmful to the correctness of color processing and needs to be fixed eventually. This PR is not particularly bad, it doesn't add any new concepts in that regard, the problem should just be more prominently visible imho.